src/App.tsx

```diff
@@ -1,26 +1,177 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useState, useEffect } from 'react';
+import { addTodo, deleteTodo, getTodos, updateTodo } from './api/todos';
 
-const USER_ID = 0;
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Errors } from './components/Errors';
+
+import { loadingObject } from './utils/loadingObject';
+import { filteredTodos } from './utils/filteredTodos';
+
+import { Todo } from './types/Todo';
+import { ErrorMessage } from './types/Errors';
+import { Loading } from './types/Loading';
+import { Filters } from './types/Filters';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Filters>(Filters.All);
+  const [loadingId, setLoadingId] = useState<Loading>({});
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => setErrorMessage(''), 3000);
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.UnableToLoad);
+        clearTimeout(timeoutId);
+      });
+
+    return () => clearTimeout(timeoutId);
+  }, []);
+
+  const handleAdd = (newTodo: Todo): Promise<Todo | void> => {
```

Comment:
why u need Todo in this place?
you don't use this feature to get an item(onSubmit - ur logic)
```suggestion
  const handleAdd = (newTodo: Todo): Promise<void> => {
```